app/javascript/article-form/articleForm.jsx

```diff
@@ -133,20 +171,51 @@ export default class ArticleForm extends Component {
     });
   };
 
+  removeLocalStorage = () => {
+    localStorage.removeItem(this.url);
+    window.removeEventListener('beforeunload', this.localStoreContent);
+  };
+
   onPublish = e => {
     e.preventDefault();
     this.setState({ submitting: true, published: true });
     const state = this.state;
     state.published = true;
-    submitArticle(state, this.handleArticleError);
+    submitArticle(state, this.removeLocalStorage, this.handleArticleError);
   };
 
   onSaveDraft = e => {
     e.preventDefault();
     this.setState({ submitting: true, published: false });
     const state = this.state;
     state.published = false;
-    submitArticle(state, this.handleArticleError);
+    submitArticle(state, this.removeLocalStorage, this.handleArticleError);
+  };
+
+  onClearChanges = e => {
+    e.preventDefault();
+    confirm('Are you sure you want to revert to the previous save?');
```

Comment:
`confirm` returns a boolean value so consider acting on the return value, otherwise `this.setState` will always be called, so perhaps e.g.

```
if (!confirm('Are you sure you want to revert to the previous save?')) {
 return;
}

this.setState ....
```

Reply:
Great catch! Thanks!